### PR TITLE
Added Siamese Model

### PR DESCRIPTION
**Pull request by AlekseySh**

### CODE REVIEW

**Review by DaloroAT — APPROVED** [review]

File: oml/models/siamese.py [hunk 1]
```diff
@@ -45,37 +50,107 @@ def forward(self, x1: Tensor, x2: Tensor) -> Tensor:
...
+
+        # turn off the last bias
+        self.head[-2] = nn.Linear(self.head[-2].in_features, self.head[-2].out_features, bias=False)
+
+        # turn off the last dropout
+        self.head[-1] = nn.Identity()
+
+        self.train_backbone = True
+
+        if weights:
```
[the remark this answers is not in the document]
Comment (AlekseySh) [reply oml/models/siamese.py:96]: discussed offline

File: oml/registry/models.py [hunk 2]
```diff
@@ -23,15 +34,19 @@ def raise_if_needed(extractor_cfg: TCfg, kwargs: Dict[str, Any], model_name: str
...
+        model = registry[model_name](extractor=inside_extractor, **kwargs)  # type: ignore
         raise_if_needed(extractor_cfg, kwargs, model_name)
     else:
-        extractor = MODELS_REGISTRY[model_name](**kwargs)  # type: ignore
-    return extractor
+        model = registry[model_name](**kwargs)  # type: ignore
+    return model
+
+
+def get_extractor(model_name: str, **kwargs: Dict[str, Any]) -> IExtractor:
```
[the remark this answers is not in the document]
Comment (AlekseySh) [reply oml/registry/models.py:48]: discussed offline

File: oml/models/siamese.py [hunk 3]
```diff
@@ -45,37 +50,107 @@ def forward(self, x1: Tensor, x2: Tensor) -> Tensor:
...
-        super(ResNetSiamese, self).__init__()
-        self.backbone = resnet18(pretrained=pretrained)
+        super(ConcatSiamese, self).__init__()
+        self.extractor = extractor
+
+        self.head = MLP(
+            in_channels=self.extractor.feat_dim,
+            hidden_channels=[*mlp_hidden_dims, 1],
+            activation_layer=Sigmoid,
+            dropout=0.5,
```
Comment (dapladoc) [inline oml/models/siamese.py:84]: Maybe it makes sense to put the `dropout` parameter to the correspondent yaml?

**Review by dapladoc — COMMENTED** [review]
I have just one small suggestion/comment. The rest looks good.

**Comment by AlekseySh** [comment]
@dapladoc thank you for the review! I guess after a series of experiments we return to the model and rework it, including parameters handling

File: oml/models/siamese.py (hunk 3 above)
Comment (AlekseySh) [reply oml/models/siamese.py:84]: @dapladoc hank you for the review! I guess after a series of experiments we return to the model and rework it, including parameters handling
